content/en/blog/_posts/2019-06-13-volume-cloning-alpha.md

```diff
@@ -0,0 +1,114 @@
+
+By providing a standard way to trigger clone operations in the Kubernetes API, Kubernetes users can now handle use cases like this without having to go around the Kubernetes API (and manually executing storage system specific operations).
+
+Instead, Kubernetes users are now empowered to incorporate clone operations in a cluster agnostic way into their tooling and policy with the comfort of knowing that it will work against arbitrary Kubernetes clusters regardless of the underlying storage.
+
+## Which volume plugins support Kubernetes Cloning?
+
+Kubernetes supports three types of volume plugins: in-tree, Flex, and CSI. See [Kubernetes Volume Plugin FAQ](https://github.com/kubernetes/community/blob/master/sig-storage/volume-plugin-faq.md) for details.
+
+Cloning is only supported for CSI drivers (not for in-tree or Flex). To use the Kubernetes cloning feature, ensure that a CSI Driver that implements cloning is deployed on your cluster.
+
+## Kubernetes API and Cloning
+
+The cloning feature in Kubernetes is an enhancement to the existing PersistentVolumeClaim objects and it's API.  There are no new objects introduced to enable cloning, instead the existing DataSource field in the PersistentVolumeClaim object is expanded to be able to accept the name of an existin PersistentVolumeClaim in the users namespace.  It is important to note that from a users perspective a clone is just another Volume, the only difference being that that volume is being populated with the contents of another volume at creation time.  After creation it behaves exactly like any other Kubernetes Volume and adhers to the same behaviors and rules as any other Volume.
+
+## Kubernetes Cloning Requirements
+
+Before using Kubernetes Volume Cloning, you must:
+
+* Ensure a CSI driver implementing Cloning is deployed and running on your Kubernetes cluster.
+* Enable the Kubernetes Volume Cloning feature via new Kubernetes feature gate (disabled by default for alpha):
+  * Set the following flag on the API server binary: `--feature-gates=VolumePVCDataSource=true`
+
+## Creating a clone with Kubernetes
+
+To provision a new volume pre-populated with data from an existing Kubernetes Volume, use the dataSource field in the `PersistentVolumeClaim`.  There are three parameters:
+
+* name - name of the `PersistentVolumeClaim` object to use as source
+* kind - must be `PersistentVolumeClaim`
+* apiGroup - must be `""`
+
+The namespace of the source `PersistentVolumeClaim` (source) object is assumed to be the same as the namespace of the new `PersistentVolumeClaim` (dstination) being created object.
+
+```
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-clone
+  Namespace: demo-namespace
+spec:
+  storageClassName: csi-storageclass
+  dataSource:
+    name: src-pvc
+    kind: PersistentVolumeclaim 
+    apiGroup: ""
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
```

Comment:
does capacity matter?

Reply:
Yeah, the idea being to allow automatic expansion on clone.  At least for now the implementation requires capacity be requested.

Reply:
So the capacity has to be  >= the source PV?